storages/replicator.go

```diff
@@ -28,8 +29,14 @@ func newReplicationClient(backends []*backend.Backend) client {
 func (rc *ReplicationClient) Do(request *http.Request) <-chan BackendResponse {
 	responsesChan := make(chan BackendResponse)
 	wg := sync.WaitGroup{}
-	ctx, cancelFunc := context.WithCancel(request.Context())
+	reqIDValue, ok := request.Context().Value(log.ContextreqIDKey).(string)
+	if !ok {
+		reqIDValue = ""
```

Comment:
Why can't we generate an id if it's not in the context?

Reply:
To distinguish between lost request ID and legit request ID